### PR TITLE
Fix logic errors in TreeNode.DeepCopy

### PR DESCRIPTION
**Pull request by raararaara**
**What this PR does / why we need it**:

Fix logic errors in TreeNode.DeepCopy

This PR addresses the following logic errors in `TreeNode.DeepCopy`:
1. Missing insPrevID and insNextID: resolved in https://github.com/yorkie-team/yorkie-js-sdk/pull/756 and ported accordingly.
2. Incorrect calculation of IndexTreeNode.Length - including tombstone nodes in length calculation.

This commit fixes these issues, and corresponding test cases have been added.

**Which issue(s) this PR fixes**:

Address [#754](https://github.com/yorkie-team/yorkie-js-sdk/issues/754)
Related to https://github.com/yorkie-team/yorkie-js-sdk/pull/756

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Additional documentation**:

```docs

```

**Checklist**:
- [x] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: pkg/index/tree.go [hunk 1]
```diff
@@ -319,7 +319,9 @@ func (n *Node[V]) SetChildren(children []*Node[V]) error {
 	n.children = children
 	for _, child := range children {
 		child.Parent = n
-		child.UpdateAncestorsSize()
+		if !child.Value.IsRemoved() {
+			child.UpdateAncestorsSize()
+		}
```
Comment (hackerwins) [inline pkg/index/tree.go:324]: This change does not exist in the PR of JS SDK. Could you explain?
https://github.com/yorkie-team/yorkie-js-sdk/pull/756

File: test/integration/tree_test.go [hunk 2]
```diff
@@ -792,6 +792,225 @@ func TestTree(t *testing.T) {
...
+
+		assert.NoError(t, c3.Sync(ctx))
+		assert.NoError(t, c2.Sync(ctx))
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 2 네이3</n></p></c></r>", d3.Root().GetTree("t").ToXML())
+		assert.Equal(t, "<r><c><u><p><n></n></p></u></c><c><p><n>1 2 네이3</n></p></c></r>", d2.Root().GetTree("t").ToXML())
+
+		assert.NoError(t, c3.Deactivate(context.Background()))
+		assert.NoError(t, c3.Close())
+	})
+
```
Comment (hackerwins) [inline test/integration/tree_test.go:1013]: I remember that we will be working on simplifying the test code. Will you work on the next PR?

**Review by hackerwins — COMMENTED** [review]
Thanks for your contribution.
I left simple questions.

File: test/integration/tree_test.go (hunk 2 above)
Comment (raararaara) [reply test/integration/tree_test.go:1013]: Of course, I will simplify it and reflect it in the next PR.

File: pkg/index/tree.go (hunk 1 above)
Comment (raararaara) [reply pkg/index/tree.go:324]: In js-sdk, crdtTreeNode inherits indexTreeNode. That's why deepcopy in [js-sdk can calculate the size](https://github.com/yorkie-team/yorkie-js-sdk/blob/2af8a0c791667f7143eb72599a5c8c4a9f9ecd76/src/document/crdt/tree.ts#L451) right away.
However in the case of go-sdk, there is no such structure, so it seemed necessary to update the size (Length in go-sdk) of the indexTreeNode for ancestors during the process of `setChildren()`.

Comment (raararaara) [reply pkg/index/tree.go:324]:
```go
func (n *TreeNode) DeepCopy() (*TreeNode, error) {
        // ...
	clone.RemovedAt = n.RemovedAt
	clone.Index.Length = n.Index.Length
	clone.InsPrevID = n.InsPrevID
	clone.InsNextID = n.InsNextID

	if n.IsText() {
		return clone, nil
	}
        // ...
}

// SetChildren sets the children of the given node.
func (n *Node[V]) SetChildren(children []*Node[V]) error {
	if n.IsText() {
		return ErrInvalidMethodCallForTextNode
	}

	n.children = children
	for _, child := range children {
		child.Parent = n
		// erase here:
		// child.UpdateAncestorsSize() 
	}

	return nil
}
```

I thought of another way to solve it by directly referencing the IndexTreeNode and changing the Length to be copied as well, and not changing the Length in `setChildren()`. What do you think?

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.

**Pull request merged**